### PR TITLE
Use the Comparator Interface and Collator Class for Sorting

### PR DESCRIPTION
**Pull request by satishYadav032**
Ticket included : [AVE_BLOG79](https://theavenoir.atlassian.net/jira/software/projects/AVEBLOG/boards/4?selectedIssue=AVEBLOG-97)
Issue :Use the Comparator Interface and Collator Class for Sorting
Cause : N/A
fix: N/A

### CODE REVIEW

File: force-app/main/default/classes/CompareByStudentName.cls [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    * Constructor Purpose: Compares two StudentRecord instances based on the student's 
+      name to determine their order. 
+    * @param StudentRecord : Name1
+    * @param StudentRecord : Name2
+    */
+
+    public Integer compare(StudentRecord Name1, StudentRecord Name2) {
+        Integer returnValue;
+        if(String.isBlank(Name1.studentName) && String.isBlank(Name2.studentName)) 
+        {
```
Comment (AvenoirShiv) [inline force-app/main/default/classes/CompareByStudentName.cls:24]: Incorrect Indentation

File: force-app/main/default/classes/CompareByStudentName.cls [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            returnValue = 0;
+        }
```
Comment (AvenoirShiv) [inline force-app/main/default/classes/CompareByStudentName.cls:26]: All Indentation are wrong.

File: force-app/main/default/classes/CompareByStudentName.cls [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      name to determine their order. 
+    * @param StudentRecord : Name1
+    * @param StudentRecord : Name2
+    */
+
+    public Integer compare(StudentRecord Name1, StudentRecord Name2) {
+        Integer returnValue;
+        if(String.isBlank(Name1.studentName) && String.isBlank(Name2.studentName)) {
+            returnValue = 0;
+        } else {
```
Comment (AvenoirShiv) [inline force-app/main/default/classes/CompareByStudentName.cls:25]: Incorrect Indentation

File: force-app/main/default/classes/CompareByStudentNameTest.cls [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    */
+
+    @isTest static void testCompareSecondStudentNameBlank() {
+        StudentRecord student1 = new StudentRecord('Alice', '123456789');
+        StudentRecord student2 = new StudentRecord(null, '987654321');
+        CompareByStudentName comparator = new CompareByStudentName();
+        Integer result = comparator.compare(student1, student2);
+        // Assert
+        System.assertEquals(1, result,
+            'A non-blank student name should be considered greater than a blank name'
```
Comment (AvenoirShiv) [inline force-app/main/default/classes/CompareByStudentNameTest.cls:59]: custom label

File: force-app/main/default/classes/CompareByStudentNameTest.cls [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    @isTest static void testCompareBothNamesBlank() {
+        StudentRecord student1 = new StudentRecord(null, '123456789');
+        StudentRecord student2 = new StudentRecord(null, '987654321');
+        CompareByStudentName comparator = new CompareByStudentName();
+        Integer result = comparator.compare(student1, student2);
+        // Assert
+        System.assertEquals(0, result,
+            'Two blank student names should be considered equal'
```
Comment (AvenoirShiv) [inline force-app/main/default/classes/CompareByStudentNameTest.cls:70]: Use custom label

File: force-app/main/default/classes/StudentRecord.cls [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+* Utility Test Class: StudentRecordTest.cls
+--------------------------------------------------------------------------------
+* Version History:
+* VERSION     DEVELOPER NAME      DATE       DETAIL FEATURES
+* 1.0         Satish Yadav      05/02/2024   Initial Development
+* ****************************************************************************/
+public with sharing class StudentRecord {
+    public Id Id;
+    public String studentName;
+    public String Phone;
```
Comment (AvenoirShiv) [inline force-app/main/default/classes/StudentRecord.cls:15]: Variable name should be start by lower-case

File: force-app/main/default/classes/StudentRecordTest.cls [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    * @Annotation : @isTest
+    * @param : Na
+    */
+    
+    @isTest static void testStudentRecordConstructor() {
+        String expectedStudentName = 'John Doe';
+        String expectedPhone = '1234567890';
+        StudentRecord testStudent = new StudentRecord(expectedStudentName, expectedPhone);
+        // Assert: Verify that the fields are correctly assigned
+        System.assertEquals(expectedStudentName, testStudent.studentName, 'The student name should match the expected value.');
```
Comment (AvenoirShiv) [inline force-app/main/default/classes/StudentRecordTest.cls:26]: Use custom label

File: force-app/main/default/classes/CompareByStudentNameTest.cls [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+    */
+
+    @isTest static void testCompareFirstStudentNameBlank() {
+        StudentRecord student1 = new StudentRecord(null, '123456789');
+        StudentRecord student2 = new StudentRecord('Bob', '987654321');
+        CompareByStudentName comparator = new CompareByStudentName();
+        Integer result = comparator.compare(student1, student2);
+        // Assert
+        System.assertEquals(-1, result,
+            'A blank student name should be considered less than any non-blank name'
```
Comment (AvenoirShiv) [inline force-app/main/default/classes/CompareByStudentNameTest.cls:42]: Use custom label

File: force-app/main/default/classes/CompareByStudentNameTest.cls [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+     * @param : Na
+    */
+
+    @isTest static void testCompareBothNamesPresent() {
+        StudentRecord student1 = new StudentRecord('Alice', '123456789');
+        StudentRecord student2 = new StudentRecord('Bob', '987654321');
+        CompareByStudentName comparator = new CompareByStudentName();
+        Integer result = comparator.compare(student1, student2);
+        // Assert
+        System.assert(result < 0, 'Alice should come before Bob alphabetically');
```
Comment (AvenoirShiv) [inline force-app/main/default/classes/CompareByStudentNameTest.cls:26]: Use custom label

File: force-app/main/default/classes/CompareByStudentName.cls [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+public with sharing class CompareByStudentName implements Comparator<StudentRecord> {
+
+    /**
+    * Constructor Purpose: Compares two StudentRecord instances based on the student's 
+      name to determine their order. 
+    * @param StudentRecord : Name1
+    * @param StudentRecord : Name2
+    */
+
+    public Integer compare(StudentRecord Name1, StudentRecord Name2) {
```
Comment (AvenoirShiv) [inline force-app/main/default/classes/CompareByStudentName.cls:21]: Variable name should be start by lower-case

**Review by AvenoirShiv — APPROVED** [review]

File: force-app/main/default/classes/CompareByStudentName.cls [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+public with sharing class CompareByStudentName implements Comparator<StudentRecord> {
+
+    /**
+    * Constructor Purpose: Compares two StudentRecord instances based on the student's 
+      name to determine their order. 
+    * @param StudentRecord : name1
+    * @param StudentRecord : name2
+    */
+
+    public Integer compare(StudentRecord name1, StudentRecord name2) {
```
Comment (amlendu11) [inline force-app/main/default/classes/CompareByStudentName.cls:21]: update name1 and name2 to studentOne and StudentTwo

Comment (amlendu11) [reply force-app/main/default/classes/CompareByStudentName.cls:21]: Integer returnValue = -1;
        if(String.isNotBlank(name1.studentName)) {
            returnValue = String.isBlank(name2.studentName) ? 1 : name1.studentName.compareTo(name2.studentName);
        } 
        return returnValue;

Comment (amlendu11) [reply force-app/main/default/classes/CompareByStudentName.cls:21]: Refactor method like above

**Pull request merged**